meta-lmp-bsp/dynamic-layers/freescale-layer/recipes-bsp/imx-atf/imx-atf/0003-plat-imx8m-add-SiP-call-for-triggering-internal-warm.patch

```diff
@@ -0,0 +1,114 @@
+From 2d17a746addbba02f2adf4c55e850eff0479746f Mon Sep 17 00:00:00 2001
+From: Igor Opaniuk <igor.opaniuk@foundries.io>
+Date: Thu, 18 Mar 2021 20:11:51 +0200
+Subject: [PATCH 2/2] plat: imx8m: add SiP call for triggering internal warm
+ reset
+
+Add SiP call for asserting Software Reset Signal (internal reset) by
+WDOG.
+
+Signed-off-by: Igor Opaniuk <igor.opaniuk@foundries.io>
+Change-Id: Icbbf82ffab8d224880683e0d8ffdfed6a626606e
+---
+ plat/imx/common/imx_sip_handler.c     | 43 +++++++++++++++++++++++++++
+ plat/imx/common/imx_sip_svc.c         |  4 +++
+ plat/imx/common/include/imx_sip_svc.h |  7 +++++
+ 3 files changed, 54 insertions(+)
+
+diff --git a/plat/imx/common/imx_sip_handler.c b/plat/imx/common/imx_sip_handler.c
+index 903f6b8cc..69b9512f7 100644
+--- a/plat/imx/common/imx_sip_handler.c
++++ b/plat/imx/common/imx_sip_handler.c
+@@ -12,6 +12,7 @@
+ #include <common/debug.h>
+ #include <common/runtime_svc.h>
+ #include <imx_sip_svc.h>
++#include <lib/mmio.h>
+ #include <sci/sci.h>
+ #include <errno.h>
+ 
+@@ -198,6 +199,48 @@ int imx_get_cpu_rev(uint32_t *cpu_id, uint32_t *cpu_rev)
+ }
+ #endif /* defined(PLAT_imx8qm) || defined(PLAT_imx8qx) || defined(PLAT_imx8dx) || defined(PLAT_imx8qm) || defined(PLAT_imx8dxl) */
+ 
++#if defined(PLAT_imx8mq) || defined(PLAT_imx8mm)
++int imx_warm_reset_handler(uint32_t smc_fid, u_register_t x1,
```

Comment:
__attribute__((noreturn))  ?

Reply:
hm, why?

Reply:
does this function return?

Reply:
```
return SMC_OK;
```

Reply:
No, there is a while loop just above that useless return statement. IMO you should do it right.

Reply:
fixed, thanks